backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ReleaseService.java

```diff
@@ -19,22 +31,149 @@ public class ReleaseService implements ReleaseAnnouncementUseCase {
 
   private final LoadAnnouncementByNormEliUseCase loadAnnouncementByNormEliUseCase;
   private final UpdateAnnouncementPort updateAnnouncementPort;
+  private final UpdateOrSaveNormPort updateOrSaveNormPort;
+  private final NormService normService;
+  private final TimeMachineService timeMachineService;
+  private final CreateNewVersionOfNormService createNewVersionOfNormService;
+  private final DeleteQueuedNormsPort deleteQueuedNormsPort;
+  private final DeleteUnpublishedNormPort deleteUnpublishedNormPort;
 
   public ReleaseService(
     LoadAnnouncementByNormEliUseCase loadAnnouncementByNormEliUseCase,
-    UpdateAnnouncementPort updateAnnouncementPort
+    UpdateAnnouncementPort updateAnnouncementPort,
+    UpdateOrSaveNormPort updateOrSaveNormPort,
+    NormService normService,
+    TimeMachineService timeMachineService,
+    CreateNewVersionOfNormService createNewVersionOfNormService,
+    DeleteQueuedNormsPort deleteQueuedNormsPort,
+    DeleteUnpublishedNormPort deleteUnpublishedNormPort
   ) {
     this.loadAnnouncementByNormEliUseCase = loadAnnouncementByNormEliUseCase;
     this.updateAnnouncementPort = updateAnnouncementPort;
+    this.updateOrSaveNormPort = updateOrSaveNormPort;
+    this.normService = normService;
+    this.timeMachineService = timeMachineService;
+    this.createNewVersionOfNormService = createNewVersionOfNormService;
+    this.deleteQueuedNormsPort = deleteQueuedNormsPort;
+    this.deleteUnpublishedNormPort = deleteUnpublishedNormPort;
   }
 
+  /**
+   * Queue all norms that are affected by the given announcement for publishing.
+   * This includes generating a new manifestation of the amending norm, and new expressions for each time boundary of
```

Comment:
Generating a new manifestation of the amending norm even if nothing changed since it was imported?

Reply:
its needed if the amending norm was already published before, and in the case of the first publish there is no harm done by doing it anyway. It's probably actually better as then the pointInTimeManifestation is the date of the publishing.